src/pg/sql/41_observatory_augmentation.sql

```diff
@@ -0,0 +1,995 @@
+
+CREATE OR REPLACE FUNCTION OBS_GetSegmentFamiliesWithYoungChildren(the_geom geometry)
+RETURNS TABLE (
+  families_with_young_children NUMERIC,
+  two_parent_families_with_young_children NUMERIC,
+  two_parents_in_labor_force_families_with_young_children NUMERIC,
+  two_parents_father_in_labor_force_families_with_young_children NUMERIC,
+  two_parents_mother_in_labor_force_families_with_young_children NUMERIC,
+  two_parents_not_in_labor_force_families_with_young_children NUMERIC,
+  one_parent_families_with_young_children NUMERIC,
+  father_one_parent_families_with_young_children NUMERIC
+)
+AS $$
+DECLARE
+  column_ids text[];
+  q text;
+  segment_name text;
+BEGIN
+
+  -- tag used to search for columns which define segment
+  segment_name := '"us.census.segments".families_with_young_children';
+
+  -- get array of column ids according to segment defined in segment_name
+  EXECUTE
+    $query$
+    SELECT column_ids
+    FROM OBS_DESCRIBE_SEGMENT('"us.census.segments".families_with_young_children')
+    LIMIT 1;
+    $query$
+  INTO column_ids
+  USING segment_name;
+
+  IF column_ids IS NULL
+  THEN
+    RAISE EXCEPTION 'no segment information for this point';
+  END IF;
+
+  -- build query to get data in segment columns
+  q = 'WITH a As (
+    SELECT
+      column_names As names,
+      column_vals As vals
+    FROM OBS_GetSegment($1, $2)
+    ) ' ||
+    OBS_BuildSnapshotQuery(column_ids) ||
+    ' FROM a';
+
+  RETURN QUERY
+  EXECUTE
+  q
+  USING the_geom, segment_name;
+END;
+$$ LANGUAGE plpgsql;
+
+--Base functions for performing augmentation
+----------------------------------------------------------------------------------------
+
+--Creates an array-based report of the named segment given a point or geometry
+-- TODO: remove or rename since it is based on address lookups
+
+CREATE OR REPLACE FUNCTION OBS_GetSegment(
+  address text,
+  segment_name text,
+  time_span text DEFAULT '2009 - 2013',
+  geometry_level text DEFAULT '"us.census.tiger".block_group')
+RETURNS TABLE (column_names text[], column_vals numeric[])
+AS $$
+BEGIN
+
+  RETURN QUERY
+  EXECUTE
+  $query$
+    SELECT *
+      FROM OBS_GetSegment(cdb_geocode_street_point($1), $2, $3, $4);
+  $query$
+  USING address,
+        segment_name,
+        time_span,
+        geometry_level;
+  RETURN;
+END;
+$$ LANGUAGE plpgsql;
+
+
+
+CREATE OR REPLACE FUNCTION OBS_GetSegment(
+  geom geometry,
+  segment_name text,
+  time_span text DEFAULT '2009 - 2013',
+  geometry_level text DEFAULT '"us.census.tiger".block_group')
+RETURNS TABLE (column_names text[], column_vals numeric[])
+AS $$
+DECLARE
+ column_ids text[];
+BEGIN
+
+  EXECUTE
+    $query$
+    SELECT column_ids
+      FROM OBS_DESCRIBE_SEGMENT($1)
+     LIMIT 1;
+    $query$
+  INTO column_ids
+  USING segment_name;
+
+  IF column_ids IS NULL THEN
+    RAISE EXCEPTION 'Could not find segment ''%''', segment_name;
```

Comment:
I changed several to give back nulls, but there are some cases where raising exceptions makes sense, like when a polygon is passed instead of a point.